src/library_pthread.js

```diff
@@ -769,17 +777,30 @@ var LibraryPThread = {
     if (canceled == 2) throw 'Canceled!';
   },
 
+#if MINIMAL_RUNTIME
+  emscripten_check_blocking_allowed__deps: ['$warnOnce'],
+#endif
   emscripten_check_blocking_allowed: function() {
-#if ASSERTIONS
-    assert(ENVIRONMENT_IS_WEB);
-    warnOnce('Blocking on the main thread is very dangerous, see https://emscripten.org/docs/porting/pthreads.html#blocking-on-the-main-browser-thread');
+#if ASSERTIONS || IN_TEST_HARNESS || !MINIMAL_RUNTIME || !ALLOW_BLOCKING_ON_MAIN_THREAD
```

Comment:
I don't understand why `!MINIMAL_RUNTIME` is here? This code is already behind `ASSERTIONS`, so it won't be in an optimized build with the minimal runtime or not? I think it's useful in assertion builds to have maximal checks, regardless of runtime.

Reply:
Agreed, let me drop that.

Reply:
Oh, that is a `|| !MINIMAL_RUNTIME` statement and not a ` && !MINIMAL_RUNTIME` statement. The `|| !MINIMAL_RUNTIME` statement is here so that regular runtime will get the non-assertion release builds also to contain the allow blocking checks.